pkg/theia/commands/policy_recommendation_run.go

```diff
@@ -0,0 +1,473 @@
+		if err != nil || !matchResult {
+			return fmt.Errorf("executor-memory should conform to the Kubernetes convention")
+		}
+		sparkResourceArgs.executorMemory = executorMemory
+
+		kubeconfig, err := ResolveKubeConfig(cmd)
+		if err != nil {
+			return err
+		}
+		clientset, err := CreateK8sClient(kubeconfig)
+		if err != nil {
+			return fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+		}
+
+		waitFlag, err := cmd.Flags().GetBool("wait")
+		if err != nil {
+			return err
+		}
+
+		err = PolicyRecoPreCheck(clientset)
+		if err != nil {
+			return err
+		}
+
+		recommendationID := uuid.New().String()
+		recoJobArgs = append(recoJobArgs, "--id", recommendationID)
+		recommendationApplication := &sparkv1.SparkApplication{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "sparkoperator.k8s.io/v1beta2",
+				Kind:       "SparkApplication",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "policy-reco-" + recommendationID,
+				Namespace: flowVisibilityNS,
+			},
+			Spec: sparkv1.SparkApplicationSpec{
+				Type:                "Python",
+				SparkVersion:        sparkVersion,
+				Mode:                "cluster",
+				Image:               ConstStrToPointer(sparkImage),
+				ImagePullPolicy:     ConstStrToPointer(sparkImagePullPolicy),
+				MainApplicationFile: ConstStrToPointer(sparkAppFile),
+				Arguments:           recoJobArgs,
+				Driver: sparkv1.DriverSpec{
+					CoreRequest: &driverCoreRequest,
+					SparkPodSpec: sparkv1.SparkPodSpec{
+						Memory: &driverMemory,
+						Labels: map[string]string{
+							"version": sparkVersion,
+						},
+						EnvSecretKeyRefs: map[string]sparkv1.NameKey{
+							"CH_USERNAME": {
+								Name: "clickhouse-secret",
+								Key:  "username",
+							},
+							"CH_PASSWORD": {
+								Name: "clickhouse-secret",
+								Key:  "password",
+							},
+						},
+						ServiceAccount: ConstStrToPointer(sparkServiceAccount),
+					},
+				},
+				Executor: sparkv1.ExecutorSpec{
+					CoreRequest: &executorCoreRequest,
+					SparkPodSpec: sparkv1.SparkPodSpec{
+						Memory: &executorMemory,
+						Labels: map[string]string{
+							"version": sparkVersion,
+						},
+						EnvSecretKeyRefs: map[string]sparkv1.NameKey{
+							"CH_USERNAME": {
+								Name: "clickhouse-secret",
+								Key:  "username",
+							},
+							"CH_PASSWORD": {
+								Name: "clickhouse-secret",
+								Key:  "password",
+							},
+						},
+					},
+					Instances: &sparkResourceArgs.executorInstances,
+				},
+			},
+		}
+		response := &sparkv1.SparkApplication{}
+		err = clientset.CoreV1().RESTClient().
+			Post().
+			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+			Namespace(flowVisibilityNS).
+			Resource("sparkapplications").
+			Body(recommendationApplication).
+			Do(context.TODO()).
+			Into(response)
+		if err != nil {
+			return err
+		}
+		if waitFlag {
+			err = wait.Poll(statusCheckPollInterval, statusCheckPollTimeout, func() (bool, error) {
```

Comment:
if the 60min timeout expires, I assume we might still be able to check job completion via check and result commands, is that correct?

In that case it might be worth - unless too complex - to return an error message like:

Spark job wait timeout of 60minutes expired. Job <id> still running. Please check completion status for job <id> via CLI.

Reply:
Sure, added a condition for time out usecase.